added simple ssh deploy config (#1340)
JuliaDocs/Documenter.jl: docs/src/man/hosting.md

See the [`deploydocs`](@ref) function documentation for more details.

> ## Using Simple SSH

Reviewer: ```suggestion
## Using Simple SSH

```

If you do not have access to either Travis or Github for automated documentation updates,
or you are working in a private git hosting service where using these is impractical, it

> may be desirable to push objects more simply via SSH.  For such cases, simply call, e.g.

Reviewer: ```suggestion
may be desirable to push objects more simply via SSH.  For such cases, simply call, e.g.

```

```julia
deploydocs(
    repo = "github.com/USER_NAME/PACKAGE_NAME.jl.git",
    deploy_config = Documenter.SimpleSSHConfig("/path/to/ssh-key")
)
```

> If no path to the SSH key is provided, the `DOCUMENTER_KEY` environment variable will be used.

Reviewer: ```suggestion

If no path to the SSH key is provided, the `DOCUMENTER_KEY` environment variable will be used.
```

See the [`deploydocs`](@ref) function documentation for more details.

## Using Simple SSH

If you do not have access to either Travis or Github for automated documentation updates,
or you are working in a private git hosting service where using these is impractical, it
may be desirable to push objects more simply via SSH.  For such cases, simply call, e.g.

```julia
deploydocs(
    repo = "github.com/USER_NAME/PACKAGE_NAME.jl.git",
    deploy_config = Documenter.SimpleSSHConfig("/path/to/ssh-key")

> )

Reviewer: Should we maybe recommend guarding the `deploydocs` call here, e.g. with something like `"deploy" in ARGS && deploydocs(...)`? It would always deploy deploy when you're building docs locally at the moment?

Author: Sounds good, commit incoming.

See the [`deploydocs`](@ref) function documentation for more details.

## Using Simple SSH

If you do not have access to either Travis or GitHub for automated documentation updates,
or you are working in a private Git hosting service where using these is impractical, it
may be desirable to push objects more simply via SSH.  For such cases, simply call, e.g.

```julia
"deploy" ∈ ARGS && deploydocs(
    repo = "github.com/USER_NAME/PACKAGE_NAME.jl.git",
    deploy_config = Documenter.SimpleSSHConfig("/path/to/ssh-key")
)
```

> Note that calling `deploydocs` with the `SimpleSSHConfig` will always deploy, so if you
> are worried about pushing to the remote Git repository you may want to guard against it
> by, as in this example, checking if the argument `"deploy"` was passed on the command line.

Reviewer: Small rephrasing suggestion:

```suggestion
Note that calling `deploydocs` with the `SimpleSSHConfig` will always deploy. So, to avoid
unwanted accidental deployments, it is good practice to guard the `deplodocs` call, e.g. by
checking whether `"deploy"` was passed as a command line argument to the `make.jl` script
(as is done in the example above).
```
